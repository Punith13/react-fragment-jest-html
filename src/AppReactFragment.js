import React, { Component } from 'react'
import logo from './logo.svg'
import './App.css'

class AppReactFragment extends Component {
  render() {
    return (
      <React.Fragment>
        <div className="App">
          <header className="App-header">
            <p>
              Edit <code>src/App.js</code> and save to reload.
            </p>
            <a
              className="App-link"
              href="https://reactjs.org"
              target="_blank"
              rel="noopener noreferrer"
            >
              Learn React
            </a>
          </header>
        </div>
        <button className="button">Test button</button>
      </React.Fragment>
    )
  }
}

export default AppReactFragment
