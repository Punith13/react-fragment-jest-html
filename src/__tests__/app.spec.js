import React from 'react'
import renderer from 'react-test-renderer'
import AppReactFragment from '../AppReactFragment'
import App from '../App'

describe('Test App with React Fragment', () => {
  it('renders app', () => {
    const tree = renderer.create(<AppReactFragment />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('Test App without React Fragment', () => {
  it('renders app', () => {
    const tree = renderer.create(<App />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
