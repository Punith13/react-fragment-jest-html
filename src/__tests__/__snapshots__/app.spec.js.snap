// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App with React Fragment renders app 1`] = `
Array [
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        className="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
------------HTML PREVIEW---------------
<div class="App">
    <header class="App-header">
        <p>
            Edit 
            <code>
                src/App.js
            </code>
             and save to reload.
        </p>
        <a
            class="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
        >
            Learn React
        </a>
    </header>
</div>,
  <button
    className="button"
  >
    Test button
  </button>
------------HTML PREVIEW---------------
<button class="button">
    Test button
</button>,
]
`;

exports[`Test App without React Fragment renders app 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <button
    className="button"
  >
    Test button
  </button>
</div>
------------HTML PREVIEW---------------
<div class="App">
  <header class="App-header">
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      class="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <button class="button">
    Test button
  </button>
</div>
`;
